### ISSUE TITLE
 Main Calendar Design Ideas

### ISSUE THREAD

**Issue by Restuta** [Restuta/rcn.io#23]
--
Umbrella #22

**Comment by Restuta** [comment]
![](http://media1.giphy.com/media/rnub7x1hzAxAk/giphy.gif)

**Comment by Restuta** [comment]
When it's an end and beginning of the month it's easy to get lost what are you looking at. Trello solves this by assigning short month names to dates at edges, see screenshot: 
<img width="752" alt="trello-calendar" src="https://cloud.githubusercontent.com/assets/377261/11048302/f917ac02-86eb-11e5-937e-35d12cec97cb.png">

**Comment by Restuta** [comment]
## Idea 2 –  sunrise/sunset in calendar

See #24

**Comment by Restuta** [comment]
## Idea 3 – Minimap of the year

Sometimes I feel lost when I scroll calendar, I think it would be nice to show minimap of the entire year on the side, like sublime does for documents:

<img width="460" alt="2015-11-09_1441" src="https://cloud.githubusercontent.com/assets/377261/11049006/02f24fe4-86f0-11e5-85b4-3bdcda26a9d1.png">

**Comment by Restuta** [comment]
## Idea 4 – Year view of events

Like this (Fantastical):

<img width="1337" alt="2015-11-09_1447" src="https://cloud.githubusercontent.com/assets/377261/11049112/d4ad2874-86f0-11e5-9340-b9a0d2014d64.png">

**Comment by Restuta** [comment]
## Idea 5 – "Days from now"

It might be useful to show a tooltip (or some other way) when a user hovers a day in the calendar that says "8 days from today". This gives more context and better feel on  how far is something is future or how far back in was in the past.

**Comment by Me4ta** [comment]
*Idea 4 – Year view of events*
I think in our calendar we can show the number of events for each day in the year view. I mean don't use color for it, but numbers (if this different colors mean some intensity or load)

**Comment by Restuta** [comment]
@Me4ta yes colors mean intensity. We may use a different representation of amount of events, but I think it will be hard if it's going to be another number, since we already have "a number" in every cell, it's that day's calendar number. So fitting two numbers in a cell is hard. 

But, apart from color (heat-map) we can take an approach Fantastical takes:

<img width="294" alt="2015-11-12_1748" src="https://cloud.githubusercontent.com/assets/377261/11136695/b4cef740-8965-11e5-9322-60e448596121.png">

Those dot's represent events and color represent a particular calendar that even belong to.

**Comment by Me4ta** [comment]
Oh, I really like this idea with dots. It's amazing, simple, clear and useful.

**Comment by Restuta** [comment]
## Idea 6 – Weather and Rain Notifications

It's useful to know the weather for upcoming events and get notifications if there is high chance of rain or strong wind. For some people rain is a "no" factor and therefore it's important to get notified about somewhat extreme weather conditions. 

E.g. her here is how Outlook for Web shows weather:

![2015-11-23_1530](https://cloud.githubusercontent.com/assets/377261/11353383/72ebf036-91f7-11e5-8eaf-4410b1731d7b.png)

**Comment by Restuta** [comment]
## Idea 7 - (small) UI element to show duration
We can use it to show duration between events or between different fields for the same event or for anything really that requires duration 

<img width="247" alt="2015-12-01_1538" src="https://cloud.githubusercontent.com/assets/377261/11517920/01643378-9842-11e5-8ba6-9175f6555fc1.png">

**Comment by Restuta** [comment]
## Idea 8 - Living guide that follows scrolling
We can use similar idea for navigation, maybe vertical one. This theoretically might allow for better "where I am now " solution 

![image](https://cloud.githubusercontent.com/assets/377261/11648178/fea0336a-9d25-11e5-93c4-5dbf7c410b4d.png)

http://rosemarybutcher.com/

**Comment by Restuta** [comment]
## Idea 9 - Use different colors for different months and seasons
This would allow people to visually distinguish between month while scrolling. Take a look at how Basecamp 3 approaches this:

![2015-12-21_2100](https://cloud.githubusercontent.com/assets/377261/11948196/12c6fc1a-a826-11e5-899e-fc1d6afc0bff.png)

**Comment by Restuta** [comment]
## Idea 10 Year Scroll Progress indicator
--

Similar to Reading Indicator here https://codyhouse.co/demo/reading-progress-indicator/index.html

**Comment by Restuta** [comment]
## Idea - 11 Quick filtering of Events with Hanging Filters

Allow quick filtering while scrolling calendar with hanging buttons like here:
![2016-02-15_1248](https://cloud.githubusercontent.com/assets/377261/13059920/837e178a-d3e2-11e5-9eb1-094cad7bffc7.png)

It gives minimal distractions and can provide access to frequently used filtering like "crits or TT's or non-competitive events"

**Comment by Restuta** [comment]
## Idea 11 — Showing on top of a calendar next upcoming race, e.g. here is now meetup.com does it:

<img width="1201" alt="2016-02-18_1242" src="https://cloud.githubusercontent.com/assets/377261/13157548/565c1e98-d63d-11e5-9171-b6709b877abf.png">

**Comment by Restuta** [comment]
## Idea 12 – Horizontal view for days for mobile

Inspired by week view in [Calendars App](https://itunes.apple.com/us/app/calendars-by-readdle-sync/id608834326?mt=8)

![img_0863](https://cloud.githubusercontent.com/assets/377261/13189815/9926c2b0-d70e-11e5-8d8d-63775ed17146.PNG)

**Comment by Restuta** [comment]
## Idea 13 - Show two events in one row for wide screens to use pace more efficiently.
On L and XL screens we don't use space that great, but we can if we would show two events in a row in wide columns for calendars that have majority of events on Sat-Sun:

<img width="1419" alt="2016-04-04_1814" src="https://cloud.githubusercontent.com/assets/377261/14268571/55b3b2d0-fa91-11e5-923f-ab56ee93f541.png">

This way we use 50-70% of screen space, compare this to normal layout:

<img width="1404" alt="2016-04-04_1817" src="https://cloud.githubusercontent.com/assets/377261/14268599/93ea7a0c-fa91-11e5-94ef-0e80d4a898e6.png">

**Comment by Restuta** [comment]
## Idea 14 — Make our main page to have scrolling inline 
(more like training peaks calendar and Trellos, so we don't have to rely on sticky headers). 

This would require some absolute positioning and important features are:
* [ ] have a frozen top navbar and header
* [ ] have scrolling outside of container and taking full screen space
* [ ] make the entire area scroll, so there is no chance to miss scroll target 

This would require:
* [ ] setting `overflow:hidden` on `body`
* [ ] absolutely positioning calendar

**Comment by Restuta** [comment]
## Idea 15 — Compact view
Give an option to view calendar in "Compact mode" when every Cars has fixed height of 4 (two lines of text):

<img width="434" alt="compact-view" src="https://cloud.githubusercontent.com/assets/377261/18064954/a5c3c0c6-6de6-11e6-9f36-e82ea12fe352.png">

TBD what clicking on a card should do? Expand a card or show Event Details? I think it should be consistent and show Event Details but each card can have an [icon](https://design.google.com/icons/#ic_keyboard_arrow_down) to expand

**Comment by Restuta** [comment]
## Idea 16 — Calendar page layout inspired by google

<img width="1144" alt="00000498" src="https://user-images.githubusercontent.com/377261/36705138-7ae6d8ee-1b18-11e8-8ed0-1f7a593bbfeb.png">


see it live https://trends.google.com/trends/

--

Could present a calendar in a more modern way and give space for controls and UI elements
